cli/azd/pkg/auth/manager.go

```diff
@@ -538,13 +556,20 @@ func (m *Manager) LoginInteractive(
 	return newAzdCredential(m.publicClient, &res.Account, m.cloud), nil
 }
 
-func (m *Manager) LoginWithOneAuth(ctx context.Context, tenantID string, scopes []string) error {
-	if len(scopes) == 0 {
-		scopes = m.LoginScopes()
+func (m *Manager) LoginWithOneAuth(ctx context.Context, tenantID string, scopes []string, osAccountOnly bool) error {
```

Comment:
Would `silent bool` be as good as `osAccountOnly` here?

It feels that `auth` doesn't know about `oneauth`'s implementation details of os account at this level. If instead it's an important semantic difference, maybe we need `oneauth` to expose that more clearly.

Reply:
`osAccountOnly` is my attempt to express what the parameter does without assuming the reader understands OneAuth idioms. I think `silent` is reasonable in that it's the term OneAuth uses but in my mind it's jargon and I don't expect readers to understand what it means here.

Maybe this should be `brokerAccountOnly` in anticipation of auth brokers on other platforms which aren't OS components?

Reply:
Option 1 seems fine to me, I'll do that. I don't like option 2 because I think it's important to understand that in addition to not prompting for user interaction, signing in silently means signing in only an account provided by the system authentication broker e.g. the active Windows account.